src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonInEnumerationCheck.java

```diff
@@ -0,0 +1,116 @@
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Checks if unnecessary semicolon is in enum definitions.
+ * Semicolon is not needed if enum body is empty.
+ * The check will flag the following with warnings:
+ * </p>
+ * <pre>
+ * enum One {
+ *     A,B; // violation
+ * }
+ * enum Two {
+ *     A,B,; // violation
+ * }
+ * enum Three {
+ *     A,B(); // violation
+ * }
+ * enum Four {
+ *     A,B{}; // violation
+ * }
+ * enum NoSemicolon {
+ *     A,B // OK
+ * }
+ * enum Normal {
+ *     A,
+ *     B
+ *     ; // required ";", no violation
+ *     Normal(){}
+ * }
+ * </pre>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;UnnecessarySemicolonInEnumeration&quot;/&gt;
+ * </pre>
+ *
+ * @since 8.22
+ */
+@StatelessCheck
+public final class UnnecessarySemicolonInEnumerationCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_SEMI = "unnecessary.semicolon";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.ENUM_CONSTANT_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final DetailAST semicolon = getSemicolonSibling(ast);
+        if (semicolon != null && hasCodeElementsAfter(semicolon)) {
+            log(semicolon, MSG_SEMI);
+        }
+    }
+
+    private static DetailAST getSemicolonSibling(DetailAST ast) {
+        DetailAST result = null;
+        DetailAST curr = ast;
+        while (curr.getNextSibling() != null) {
+            final DetailAST nextSibling = curr.getNextSibling();
+            if (nextSibling.getType() == TokenTypes.SEMI) {
+                result = nextSibling;
+                break;
+            }
+            curr = nextSibling;
+        }
+        return result;
+    }
+
+    private static boolean hasCodeElementsAfter(DetailAST ast) {
```

Comment:
Should this method name be something like `hasNoCodeElementsAfter`?
It does the opposite of its name.

Reply:
There used to be a negation, but after refactoring I forgot to change method name.

Thank you, corrected.